docs/mkdocs/docs/integration/migration_guide.md

```diff
@@ -0,0 +1,248 @@
+  [`json_pointer::operator==`](../api/json_pointer/operator_eq.md) or
+  [`json_pointer::operator!=`](../api/json_pointer/operator_ne.md).
+    
+    === "Deprecated"
+  
+        ```cpp
+        nlohmann::json::json_pointer lhs("/foo/bar/1");
+        assert(lhs == "/foo/bar/1");
+        ```
+  
+    === "Future-proof"
+
+        ```cpp
+        nlohmann::json::json_pointer lhs("/foo/bar/1");
+        assert(lhs == nlohmann::json::json_pointer("/foo/bar/1"));
+        ```
+
+- The implicit conversion from JSON Pointers to string
+  ([`json_pointer::operator string_t`](../api/json_pointer/operator_string_t.md)) is deprecated since 3.11.0. Use
+  [`json_pointer::to_string`](../api/json_pointer/to_string.md) instead.
+  
+    === "Deprecated"
+  
+          ```cpp
+          nlohmann::json::json_pointer ptr("/foo/bar/1");
+          std::string s = ptr;
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          nlohmann::json::json_pointer ptr("/foo/bar/1");
+          std::string s = ptr.to_string();
+          ```
+
+- Passing a `basic_json` specialization as template parameter `RefStringType` to
+  [`json_pointer`](../api/json_pointer/index.md) is deprecated since 3.11.0. The string type can now be directly
+  provided.
+  
+    === "Deprecated"
+  
+          ```cpp
+          using my_json = nlohmann::basic_json<std::map, std::vector, my_string_type>;
+          nlohmann::json_pointer<my_json> ptr("/foo/bar/1");
+          ```
+  
+    === "Future-proof"
+  
+          ```cpp
+          nlohmann::json_pointer<my_string_type> ptr("/foo/bar/1");
+          ```
+
+#### Miscellaneous functions
+
+- The function `iterator_wrapper` is deprecated since 3.1.0. Please use the member function
+  [`items`](../api/basic_json/items.md) instead.
+  
+    === "Deprecated"
+
+          ```cpp
+          for (auto &x : nlohmann::json::iterator_wrapper(j))
+          {
+              std::cout << x.key() << ":" << x.value() << std::endl;
+          }
+          ```
+
+    === "Future-proof"
+
+          ```cpp
+          for (auto &x : j.items())
+          {
+              std::cout << x.key() << ":" << x.value() << std::endl;
+          }
+          ```
+
+- Function `friend std::ostream& operator>>(const basic_json&, std::ostream&)` is deprecated since 3.0.0. Please use
+  [`friend operator<<(std::ostream&, const basic_json&)`](../api/operator_ltlt.md) instead.
+  
+    === "Deprecated"
+
+          ```cpp
+          j >> std::cout;
+          ```
+
+    === "Future-proof"
+
+          ```cpp
+          std::cout << j;
+          ```
+
+- The legacy comparison behavior for discarded values is deprecated since 3.11.0. It is already disabled by default and
+  can still be enabled by defining
+  [`JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON`](../api/macros/json_use_legacy_discarded_value_comparison.md).
+  
+    === "Deprecated"
+
+          ```cpp
+          #define JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON
```

Comment:
```suggestion
  [`JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON`](../api/macros/json_use_legacy_discarded_value_comparison.md) to `1`.
  
    === "Deprecated"

          ```cpp
          #define JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON 0
```

Reply:
This should be 1 not 0, right?

Reply:
Fixed. Thanks!